crates/mun_hir/src/utils.rs

```diff
@@ -0,0 +1,10 @@
+use std::sync::Arc;
+
+/// Helper for mutating `Arc<[T]>` (i.e. `Arc::make_mut` for Arc slices).
+/// The underlying values are cloned if there are other strong references.
+pub(crate) fn make_mut_slice<T: Clone>(a: &mut Arc<[T]>) -> &mut [T] {
+    if Arc::get_mut(a).is_none() {
+        *a = a.iter().cloned().collect();
+    }
+    Arc::get_mut(a).unwrap()
```

Comment:
```suggestion
    if let Some(a) = Arc::get_mut(a) {
        a
    } else {
        *a = a.iter().cloned().collect();
    }
```

Reply:
We need to return the value as well. So `*a = a.iter().cloned().collect();` doesnt work. Since we know we are the ownly owners at that point we can safely unwrap.

Reply:
Never mind. I missed the assignment to the original `&mut Arc`.